TramsDataApi/Controllers/V2/TrustsController.cs

```diff
@@ -37,18 +37,21 @@ public ActionResult<ApiResponseV2<TrustSummaryResponse>> SearchTrusts(string gro
 
             var (trusts, recordCount) = _searchTrusts
                 .Execute(page, count, groupName, ukPrn, companiesHouseNumber);
-            
+
             _logger.LogInformation(
                 "Found {count} trusts for groupName \"{name}\", UKPRN \"{prn}\", companiesHouseNumber \"{number}\", page {page}, count {count}",
                 trusts.Count(), groupName, ukPrn, companiesHouseNumber, page, count);
-            
-            _logger.LogDebug(JsonSerializer.Serialize(trusts));
-            
+
+            if (_logger.IsEnabled(LogLevel.Debug))
```

Comment:
I would have thought that LogDebug will only log if the log level is set as Debug or higher.

Reply:
It does, but it will always do the JsonSerilializer.Serialize call before getting into LogDebug.

Reply:
Oh I see you think serializing the object is what's slowing this endpoint.
Its worth a shot but I think the problem is the database is being throttled by Azure because of the pricing model we're on. To get the best performance improvement we probably need to update the code to make the requests to the db in batches e.g. get all the trusts, get all the establishments and merge them in memory.

Reply:
Absolutely. It will make little difference overall.